File: langs/cowsay.hs
import System.Environment
main = do
  a <- getArgs
  let m = if null a then "Hello, World!" else unwords a
  let n = replicate (length m + 2)
  putStr $ " " ++ n '_' ++ "\n< " ++ m ++ " >\n " ++ n '-' ++ "\n        \\   ^__^\n         \\  (oo)\\_______\n            (__)\\       )\\/\\\n                ||----w |\n                ||     ||\n"

File: langs/cowsay.cpp
#include <cstdio>
#include <string>
int main(int c,char**v){
    std::string m;
    for(int i=1;i<c;i++){if(i>1)m+=" ";m+=v[i];}
    if(m.empty())m="Hello, World!";
    std::string b(m.size()+2,'_'),d(m.size()+2,'-');
    printf(" %s\n< %s >\n %s\n        \\   ^__^\n         \\  (oo)\\_______\n            (__)\\       )\\/\\\n                ||----w |\n                ||     ||\n",b.c_str(),m.c_str(),d.c_str());
}
